src/server/scripts/DragonIsles/AberrusTheShadowedCrucible/boss_kazzara_the_hellforged.cpp

```diff
@@ -93,10 +274,448 @@ struct boss_kazzara_the_hellforged : public BossAI
+        {
+            Player* player = target->ToPlayer();
+
+            if (!player)
+                return true;
+
+            if (player->HasAura(SPELL_DREAD_RIFT_SELECTOR))
+                return true;
+
+            return false;
+        });
+    }
+
+    void HandleDreadRift(SpellEffIndex /*effIndex*/)
+    {
+        GetCaster()->CastSpell(GetHitUnit(), SPELL_DREAD_RIFT_AURA, true);
+    }
+
+    void HandleDreadRiftSelector(SpellEffIndex /*effIndex*/)
+    {
+        GetCaster()->CastSpell(GetHitUnit(), SPELL_DREAD_RIFT_SELECTOR, true);
+    }
+
+    void Register() override
+    {
+        OnObjectAreaTargetSelect += SpellObjectAreaTargetSelectFn(spell_kazzara_dread_rift_player_select::FilterTargets, EFFECT_ALL, TARGET_UNIT_SRC_AREA_ENEMY);
+        OnEffectHitTarget += SpellEffectFn(spell_kazzara_dread_rift_player_select::HandleDreadRift, EFFECT_0, SPELL_EFFECT_DUMMY);
+        OnEffectHitTarget += SpellEffectFn(spell_kazzara_dread_rift_player_select::HandleDreadRiftSelector, EFFECT_1, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 407196 - Dread Rifts
+class spell_kazzara_dread_rifts : public SpellScript
+{
+    void HandleHit(SpellEffIndex /*effIndex*/)
+    {
+        uint32 spellId = SPELL_DREAD_RIFTS_3_TARGETS;
+
+        Unit::AuraApplicationMap const& appliedAuras = GetCaster()->GetAppliedAuras();
+        for (auto const& itr : appliedAuras)
+        {
+            Aura const* aura = itr.second->GetBase();
+            if (aura->GetId() == SPELL_INFERNAL_HEART_40_PCT)
+            {
+                spellId = SPELL_DREAD_RIFTS_6_TARGETS;
+                break;
+            }
+            else if (aura->GetId() == SPELL_INFERNAL_HEART_60_PCT)
+            {
+                spellId = SPELL_DREAD_RIFTS_5_TARGETS;
+                break;
+            }
+            else if (aura->GetId() == SPELL_INFERNAL_HEART_80_PCT)
+            {
+                spellId = SPELL_DREAD_RIFTS_4_TARGETS;
+                break;
+            }
+        }
+
+        GetCaster()->CastSpell(GetCaster(), spellId, true);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_kazzara_dread_rifts::HandleHit, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 402538 - Energize
+class spell_kazzara_energize : public AuraScript
+{
+    void PeriodicTick(AuraEffect const* aurEff)
+    {
+        Spell* currentChanneledSpell = GetTarget()->GetCurrentSpell(CURRENT_CHANNELED_SPELL);
+
+        if (currentChanneledSpell && (currentChanneledSpell->GetSpellInfo()->Id == SPELL_HELLSTEEL_CARNAGE_80_PCT
+            || currentChanneledSpell->GetSpellInfo()->Id == SPELL_HELLSTEEL_CARNAGE_60_PCT
+            || currentChanneledSpell->GetSpellInfo()->Id == SPELL_HELLSTEEL_CARNAGE_40_PCT))
+            return;
+
+        // Blizzard seems to follow this sequence to energize Kazzara
+        static constexpr std::array<uint8, 30> energize = { 0, 3, 4, 3, 4, 3, 4, 3, 4, 3, 4, 3, 4, 4, 4, 4, 3, 4, 3, 4, 3, 3, 4, 4, 3, 4, 3, 4, 3, 1 };
```

Comment:
@mdX7 will change this to an upper case definition and above void PeriodicTick. I have no doubts.